packages/x-components/src/x-modules/history-queries/components/__tests__/my-history.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { getXComponentXModuleName, isXComponent } from '../../../../components/index';
+import { RootXStoreState } from '../../../../store/index';
+import { DeepPartial } from '../../../../utils/index';
+import { historyQueriesXModule } from '../../x-module';
+import MyHistory from '../my-history.vue';
+import { resetXHistoryQueriesStateWith } from './utils';
+
+function renderMyHistory({
+  template = '<MyHistory />',
+  historyQueries = []
+}: MyHistoryOptions = {}): MyHistoryAPI {
+  const localVue = createLocalVue();
+  localVue.use(Vuex);
+
+  const store = new Store<DeepPartial<RootXStoreState>>({});
+  installNewXPlugin({ store, initialXModules: [historyQueriesXModule] }, localVue);
+  resetXHistoryQueriesStateWith(store, { historyQueries });
+
+  const wrapper = mount(
+    {
+      template,
+      components: {
+        MyHistory
+      }
+    },
+    {
+      localVue,
+      store,
+      propsData: {
+        historyQueries
+      }
+    }
+  );
+  return {
+    wrapper: wrapper.findComponent(MyHistory),
+    search(query) {
+      wrapper.vm.$x.emit('UserAcceptedAQuery', query);
+      return wrapper.vm.$nextTick();
+    },
+    getListItems() {
+      return wrapper.findAll(getDataTestSelector('suggestion-item'));
+    },
+    findAllInWrapper(selector) {
+      return wrapper.findAll(getDataTestSelector(selector));
+    }
+  };
+}
+
+describe('testing MyHistory component', () => {
+  it('is an XComponent which has an XModule', () => {
+    const { wrapper } = renderMyHistory();
+    expect(isXComponent(wrapper.vm)).toEqual(true);
+    expect(getXComponentXModuleName(wrapper.vm)).toEqual('historyQueries');
+  });
+
+  it('does not render the component if history is empty', () => {
+    const { wrapper } = renderMyHistory();
+    expect(wrapper.html()).toEqual('');
+  });
+
+  it('renders the list of searched queries', async () => {
+    const { search, getListItems } = renderMyHistory();
+
+    await search('lego');
+    const suggestionsWrappers = getListItems();
+    expect(suggestionsWrappers.wrappers).toHaveLength(1);
+    expect(suggestionsWrappers.at(0).text()).toEqual('lego ✕');
+  });
+
+  describe('changing history query content', () => {
+    it('allows changing history query content and render the list of history queries', () => {
+      const historyQueries = createHistoryQueries(
+        'moura',
+        'calamares',
+        'rubia galega',
+        'pulpo',
+        'cachelos',
+        'navajas',
+        'croquetas',
+        'zamburiñas'
+      );
+
+      const { findAllInWrapper } = renderMyHistory({
+        template: `
+          <MyHistory>
+            <template #suggestion-content="suggestionContentScope">
+              <img src="./history-icon.svg" data-test="suggestion-history-icon"/>
+              <span :data-index="suggestionContentScope.index"
+                    v-html="suggestionContentScope.queryHTML"></span>
```

Comment:
Test also the slot scope content, you are using it here, but you are not running any assertion over it.